src/NodeJs/lib/Writer.js

```diff
@@ -1,34 +1,200 @@
 'use strict';
 
 const { EventHubProducerClient } = require('@azure/event-hubs');
+const { MessagingError } = require('@azure/core-amqp');
+const { Transform } = require('stream');
+const binarySplit = require('binary-split');
+const UserError = require('./UserError.js');
 const ApplicationError = require('./ApplicationError.js');
+const messageStream = require('./messageStream');
+
+const MAX_BATCH_ITEMS = 1000;
+const PROGRESS_OUTPUT_INTERVAL_MS = 30 * 1000; // log progress each 30 seconds
 
 class Writer {
   constructor() {
     // Check environment variables
-    ['JSON_DELIMITER', 'CONNECTION_STRING', 'EVENT_HUB_NAME'].forEach((key) => {
+    ['MESSAGE_DELIMITER', 'CONNECTION_STRING', 'EVENT_HUB_NAME'].forEach((key) => {
       if (!process.env[key]) {
         throw new ApplicationError(`Missing "${key}" environment variable.`);
       }
     });
 
-    this.delimiter = JSON.parse(process.env.JSON_DELIMITER);
+    this.delimiter = JSON.parse(process.env.MESSAGE_DELIMITER);
     this.connectionString = process.env.CONNECTION_STRING;
     this.eventHubName = process.env.EVENT_HUB_NAME;
-    this.producer = this.createProducerClient();
+    this.progressTimer = null;
+    this.producer = null;
+    this.batch = null;
+    this.batchSendPromise = null;
+
+    // Stats
+    this.messagesQueuedCount = 0;
+    this.messagesSendingCount = 0;
+    this.messagesSentCount = 0;
+    this.batchesSentCount = 0;
   }
 
   async testConnection() {
-    await this.producer.getEventHubProperties();
+    this.producer = await this.createProducerClient();
     await this.producer.close();
   }
 
   async write() {
-    this.todo = 'todo';
+    this.startProgressTimer();
+    this.producer = await this.createProducerClient();
+    await this.writeMessages();
+    await this.producer.close();
+    this.stopProgressTimer();
+    this.logFinalState();
+  }
+
+  async writeMessages() {
```

Comment:
Implementovane ciastocne podla tohot prikladu: https://github.com/Azure/azure-sdk-for-js/blob/master/sdk/eventhub/event-hubs/samples/javascript/sendEvents.js